serpentTools/parsers/microxs.py

```diff
@@ -0,0 +1,271 @@
+                    self._storeFissionYields(chunk)
+                elif 'FLUX' in chunk[0] and self.settings['getFlx']:
+                    self._storeFluxRatio(chunk)
+                elif 'XS' in chunk[0] and self.settings['getXS']:
+                    self._storeMicroXS(chunk)
+
+    def _storeFissionYields(self, chunk):
+        """store fission yields data"""
+        fissProd, indYield, cumYield = [], [], []
+        currVar = self._strtok.search(chunk[0])
+        # Obtain the parent ID: AAZZZ0/1, e.g., 922350
+        parentFY = self._str2num(re.split('_', currVar.group())[-1 -1])
+        if 'E' in re.split('_', currVar.group())[-1]:  # e.g., NFY_902270_1E
+            sclVal = self._regexpScl.search(chunk[0])
+            # energy must be stored on the reader
+            self._energyFY = self._str2num(chunk[0][sclVal.span()[0] +
+                                                    1:sclVal.span()[1] - 2])
+            return  # thermal/epi/fast
+        else:
+            for tline in chunk:
+                if '[' in tline or ']' in tline:
+                    continue
+                if '%' in tline:
+                    tline = tline[:tline.index('%')]
+                if len(tline.split()) == 3:
+                    val1, val2, val3 = self._str2num(tline)
+                    fissProd.append(val1), indYield.append(val2), cumYield.append(val3)
+            self.nfy[(parentFY, self._energyFY)] = {'fissProd': array(fissProd),
+                                                    'indYield': array(indYield),
+                                                    'cumYield': array(cumYield)}
+
+    def _storeFluxRatio(self, chunk):
+        """store flux ratios"""
+        currVar = self._strtok.search(chunk[0])
+        # obtain the universe id
+        univ = re.split('_', currVar.group())[-1]
+        vals = self._regexpVec.search(chunk[0])  # group flux values
+        self.fluxRatio[univ], self.fluxUnc[univ] = \
+            splitItems(self._str2num(chunk[0][vals.span()[0] +
+                                              1:vals.span()[1] - 2]))
+
+    def _storeMicroXS(self, chunk):
+        """store micro cross-section and uncertainty values"""
+        currXS, currUnc = {}, {}
+        currVar = self._strtok.search(chunk[0])
+        # obtain the universe id
+        univ = re.split('_', currVar.group())[-1]
+        for tline in chunk:
+            if '[' in tline or ']' in tline:
+                continue
+            if '%' in tline:
+                tline = tline[:tline.index('%')]
+            if len(tline.split()) > 3:
+                values = self._str2num(tline)
+                # isotope, reaction type and isomeric state
+                reactionData = (values[0], values[1], values[2])
+                currXS[reactionData], currUnc[reactionData] = \
+                    splitItems(values[3:])
+        self.xsVal[univ] = currXS
+        self.xsUnc[univ] = currUnc
+
+    @staticmethod
+    def _str2num(tline):
```

Comment:
Since this is very similar to the [`ResultsReader` `_str2num`](https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/develop/serpentTools/parsers/results.py#L171) we should find a way to share the methods. This could be done two ways.
1. Move this method back to `BaseReader` 
1. Delete the occurrence back on `ResultsReader`
1. Overload the call to `str2num` for this method to perform the logical testing, e.g.
```
@staticmethod
def _str2num(line):
   vals = BaseReader._str2num(line)
   if ...
       < do a thing >
   return vals
```
or
Create a new module for these utilities that are shared by many parts of our code and import `str2num` from there